batch/batch/server/database.py

```diff
@@ -23,20 +23,43 @@ def __init__(self, db):
         super().__init__(db, 'jobs')
 
     async def update_record(self, batch_id, job_id, **items):
+        assert 'cancelled' not in items
         await super().update_record({'batch_id': batch_id, 'job_id': job_id}, items)
 
     async def get_all_records(self):
-        return await super().get_all_records()
+        async with self._db.pool.acquire() as conn:
+            async with conn.cursor() as cursor:
+                batch_name = self._db.batch.name
```

Comment:
Small aside, doesn't need to be addressed here, feels like self._db.batch.name should just be a property on self (defined in __init__)